### DISCUSSION TITLE
POI Clustering

### DISCUSSION THREAD

**Discussion by beasmm** [organicmaps/organicmaps#7991] [Ideas]
In Organic Maps, when zooming out, there is no clustering of POIs, which creates a lot of noise in the map's view and makes it hard to read. For example, in Apple Maps, the POIs are clustered into groups depending on the distance of these POIs to one another.

| OM no clustering             |  AM with clustering* |
:-------------------------:|:-------------------------:
<img src="https://github.com/organicmaps/organicmaps/assets/119599179/e1494000-05ad-458b-905f-cceab83c8c53"/>|<img src="https://github.com/organicmaps/organicmaps/assets/119599179/0d20bf80-89b9-4441-bac7-b42a217159e7"/>

*don't take into consideration the lack of POIs when zooming out, only the grouping

What do you think? If you agree, I would like to implement this feature with a friend for an assignment for our university degree in Computer Science and Engineering.

**Comment by biodranik** [comment]
That would be great, do you have experience with C++ and graphics engines?

@pastk we have already discussed clustering somewhere, right?

**Reply by pastk** [reply]
that was in some chat probably, I didn't find an issue..

**Reply by pastk** [reply]
- mentioned here https://github.com/organicmaps/organicmaps/issues/5074

There was a discussion in RU chat https://t.me/OrganicMapsRu/38298
